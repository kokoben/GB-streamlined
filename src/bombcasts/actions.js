export const BOMBCAST_VIDEO_SET = 'BOMBCAST_VIDEO_SET';
export const BOMBCAST_VIDEO_SET_SUCCESS = 'BOMBCAST_VIDEO_SET_SUCCEDED';
export const BOMBCAST_VIDEO_SET_FAIL = 'BOMBCAST_VIDEO_SET_FAIL';

export const setBombcastVideo = () => {
  console.log('inside setBombcastVideo');
  return {
    type: BOMBCAST_VIDEO_SET,
  }
}
