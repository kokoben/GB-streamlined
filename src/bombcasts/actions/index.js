import * as bActions from './types'

export const setBombcastVideo = id => {
  console.log('inside setBombcastVideo');
  return {
    type: bActions.BOMBCAST_VIDEO_SET,
    id
  }
}

export const setBombcastVideos = (page, pageSize) => {
  console.log('inside setBombcastVideos');
  return {
    type: bActions.BOMBCAST_VIDEOS_SET,
    page,
    pageSize
  }
}

export const setLatestBombcastVideo = () => ({
  type: bActions.LATEST_BOMBCAST_VIDEO_SET
})
