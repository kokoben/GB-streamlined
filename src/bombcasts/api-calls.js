// get latest bombcast video
export const requestBombcastVideo = 
  "https://www.giantbomb.com/api/videos/" +
  "?api_key=816627d452ffb34d20762fd2f3b575dfe906bfd9" +
  "&format=jsonp&json_callback=callback&limit=1&field_list=name," +
  "deck,publish_date,user,embed_player&filter=video_type:23" +
  "&sort=publish_date:desc";

export const requestBombcastVideos = page => {
  return "https://www.giantbomb.com/api/videos/" +
  "?api_key=816627d452ffb34d20762fd2f3b575dfe906bfd9" +
  "&format=jsonp&json_callback=callback&limit=24&offset=" +
  (page-1) * 24 +
  "&field_list=hd_url,name,deck,publish_date,image,video_type," +
  "user&filter=video_type:23&sort=publish_date:desc";
}
