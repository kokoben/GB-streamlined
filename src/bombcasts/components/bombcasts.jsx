import React from 'react';
import CurrentBombcastVideo from '../containers/current-bombcast-video';
import BombcastVideos from '../containers/bombcast-videos';
import FeedHeader from '../../components/feed-header';

export default () => (
  <div>
    <CurrentBombcastVideo />
    <FeedHeader
      category="Bombcast"
      placeholder="Search for Bombcast videos..."
    />
    <BombcastVideos />
  </div>
);

