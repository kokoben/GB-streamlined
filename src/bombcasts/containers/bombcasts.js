import React from 'react';
import CurrentBombcastVideo from './current-bombcast-video';
import BombcastVideos from './bombcast-videos';

export default () => (
  <div>
    <CurrentBombcastVideo />
    <BombcastVideos />
  </div>
)

