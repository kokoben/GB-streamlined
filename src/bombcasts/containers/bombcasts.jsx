import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import CurrentBombcastVideo from '../containers/current-bombcast-video';
import BombcastVideos from '../containers/bombcast-videos';
import BombcastFeedHeader from '../components/bombcast-feedheader';
import { setSearchPage } from '../../actions/index';

export class Bombcasts extends Component {
  componentDidMount() {
    this.props.setSearchPage(null);
  }

  render() {
    return (
      <div>
        <CurrentBombcastVideo />
        <BombcastFeedHeader />
        <BombcastVideos />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => (
  bindActionCreators({
    setSearchPage,
  }, dispatch)
);

Bombcasts.propTypes = {
  setSearchPage: PropTypes.func.isRequired,
};

export default connect(null, mapDispatchToProps)(Bombcasts);
