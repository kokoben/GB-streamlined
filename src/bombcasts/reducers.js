import { combineReducers } from 'redux';
import * as bActions from './actions/types';

/* eslint-disable consistent-return */
export const bombcastVideo = (state = null, action) => {
  switch (action.type) {
    case bActions.BOMBCAST_VIDEO_SET_SUCCESS:
      return action.response;
    case bActions.BOMBCAST_VIDEO_SET_FAIL:
      console.log(action.message);
      break;
    default:
      return state;
  }
};

export const bombcastVideos = (state = null, action) => {
  switch (action.type) {
    case bActions.BOMBCAST_VIDEOS_SET_SUCCESS: {
      const data = action.response;
      return data;
    }
    case bActions.BOMBCAST_VIDEOS_SET_FAIL:
      console.log(action.message);
      break;
    default:
      return state;
  }
};
/* eslint-enable */

export default combineReducers({
  bombcastVideo,
  bombcastVideos,
});
