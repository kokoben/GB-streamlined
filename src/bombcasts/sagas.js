import { put, call, takeEvery } from 'redux-saga/effects';
import jsonp from 'jsonp-promise';
import {
  BOMBCAST_VIDEO_SET,
  BOMBCAST_VIDEO_SET_SUCCESS,
  BOMBCAST_VIDEO_SET_FAIL,
  BOMBCAST_VIDEOS_SET,
  BOMBCAST_VIDEOS_SET_SUCCESS,
  BOMBCAST_VIDEOS_SET_FAIL,
  LATEST_BOMBCAST_VIDEO_SET,
  LATEST_BOMBCAST_VIDEO_SET_SUCCESS,
  LATEST_HOME_BOMBCAST_SET_FAIL
} from './actions';
import { requestBombcastVideo, requestBombcastVideos } from './api-calls';

const jsonpWrapper = args => {
  const response = jsonp(...args);
  return response.promise;
}

const params = {param: 'json_callback'};

// workers
function* setBombcastVideoAsync(action) {
  try {
    const jsonpArgs = [requestBombcastVideo(action.id), params]
    const response = yield call(jsonpWrapper, jsonpArgs);
    yield put({type: BOMBCAST_VIDEO_SET_SUCCESS, response: response.results});
  } catch (e) {
    console.log('bombcast_video_set_failed!');
    yield put({type: BOMBCAST_VIDEO_SET_FAIL, message: e.message});
  }
}

function* setLatestBombcastVideoAsync() {
  try {
    const jsonpArgs = [requestBombcastVideos(1), params];
    const response = yield call(jsonpWrapper, jsonpArgs);
    yield put({type: BOMBCAST_VIDEO_SET_SUCCESS, response: response.results[0]});
  } catch (e) {
    yield put({type: BOMBCAST_VIDEO_SET_FAIL, message: e.message});
  }
}

function* setBombcastVideosAsync(action) {
  try {
    const jsonpArgs = [requestBombcastVideos(action.page), params];
    const response = yield call(jsonpWrapper, jsonpArgs);
    yield put({type: BOMBCAST_VIDEOS_SET_SUCCESS, response: response});
  } catch (e) {
    console.log('bombcast_videos_set_failed!');
    yield put({type: BOMBCAST_VIDEOS_SET_FAIL, message: e.message});
  }
}

// watchers
export function* watchSetBombcastVideo() {
  console.log('redux-saga is running the BOMBCAST_VIDEO_SET action listener');
  yield takeEvery(BOMBCAST_VIDEO_SET, setBombcastVideoAsync);
}

export function* watchSetLatestBombcastVideo() {
  console.log('redux-saga is running the LATEST_BOMBCAST_VIDEOS_SET action listener');
  yield takeEvery(LATEST_BOMBCAST_VIDEO_SET, setLatestBombcastVideoAsync);
}

export function* watchSetBombcastVideos() {
  console.log('redux-saga is running the BOMBCAST_VIDEOS_SET action listener');
  yield takeEvery(BOMBCAST_VIDEOS_SET, setBombcastVideosAsync);
}

