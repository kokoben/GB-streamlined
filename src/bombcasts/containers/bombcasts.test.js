import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Bombcasts } from './bombcasts';

it('renders correctly', () => {
  const mockSetSearchPage = jest.fn();
  const wrapper = shallow(<Bombcasts
    setSearchPage={mockSetSearchPage}
  />);

  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
