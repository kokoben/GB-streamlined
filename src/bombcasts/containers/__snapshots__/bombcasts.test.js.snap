// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <Connect(CurrentBombcastVideo) />
  <Connect(Component) />
  <Connect(BombcastVideos) />
</div>
`;
